src/main/scala/gpp/ui/forms/ExternalValue.scala

```diff
@@ -0,0 +1,29 @@
+package gpp.ui.forms
+
+import japgolly.scalajs.react.Callback
+import crystal.ViewF
+import crystal.react.implicits._
+import cats.effect.Effect
+import japgolly.scalajs.react.extra.StateSnapshot
+
+trait ExternalValue[W[_]] {
+  def get[A](w: W[A]): A
+  def set[A](w: W[A]): A => Callback
+}
+
+object ExternalValue {
+  implicit def externalValueViewF[F[_]: Effect]: ExternalValue[ViewF[F, *]] =
+    new ExternalValue[ViewF[F, *]] {
```

Comment:
I don't know that syntax. what is the `*` in `ViewF[F, *]`?

Reply:
That's kind projector syntax. It allows you to partially apply type parameters easily.